src/python/pants/engine/internals/graph.py

```diff
@@ -1174,15 +1179,20 @@ async def resolve_dependencies(
     local_environment_name: ChosenLocalEnvironmentName,
 ) -> Addresses:
     environment_name = local_environment_name.val
-    wrapped_tgt, explicitly_provided = await MultiGet(
-        Get(
-            WrappedTarget,
-            # It's only possible to find dependencies for a target that we already know exists.
-            WrappedTargetRequest(request.field.address, description_of_origin="<infallible>"),
-        ),
-        Get(ExplicitlyProvidedDependencies, DependenciesRequest, request),
+    wrapped_tgt = await Get(
+        WrappedTarget,
+        # It's only possible to find dependencies for a target that we already know exists.
+        WrappedTargetRequest(request.field.address, description_of_origin="<infallible>"),
     )
     tgt = wrapped_tgt.target
+    try:
```

Comment:
You could wrap the whole MultiGet in the try block to preserve the parallelization.

Reply:
I wouldn't have the `tgt` for the error message then..

Reply:
I did consider doing the lookup for `tgt` in the error handler to preserve the parallelization for the normal flow, but figured that the target lookup is likely not that expensive operation so we shouldn't loose that much by not running these in parallel.

Reply:
Makes sense